pkg/gcp/client/compute.go

```diff
@@ -409,6 +415,45 @@ func (c *computeClient) ListRoutes(ctx context.Context, opts RouteListOpts) ([]*
 	return res, nil
 }
 
+func (c *computeClient) InsertAliasIPRoute(ctx context.Context, route apisgcp.Route, defaultSecondarySubnetName string) error {
+	const (
+		maxRetries     = 5
+		initialBackoff = 1 * time.Second
+		backoffFactor  = 2
+	)
+	log := shared.LogFromContext(ctx)
+
+	instance, err := c.GetInstance(ctx, route.Zone, route.InstanceName)
+	if err != nil {
+		return err
+	}
+
+	// Add an alias IP range to the first network interface
+	aliasIpRange := &compute.AliasIpRange{
+		IpCidrRange:         route.DestinationCIDR,
+		SubnetworkRangeName: defaultSecondarySubnetName,
+	}
+
+	// Append the alias IP range to the existing list
+	instance.NetworkInterfaces[0].AliasIpRanges = append(instance.NetworkInterfaces[0].AliasIpRanges, aliasIpRange)
+
+	// Retry logic for updating the instance
+	var op *compute.Operation
+	for i := 0; i < maxRetries; i++ {
+		op, err = c.service.Instances.UpdateNetworkInterface(c.projectID, route.Zone, route.InstanceName, instance.NetworkInterfaces[0].Name, instance.NetworkInterfaces[0]).Context(ctx).Do()
+		if err == nil {
+			log.Info("Operation to update network interface started: ", "opName", op.Name, "name", instance.NetworkInterfaces[0].Name)
+			return nil
+		}
```

Comment:
- Do we need a retry for this operation ?
- Do you not need to check if the operation completes successfully ? Here you only check that the operation is queued.
- I would say that maybe extract the exp.Retry in it's own func instead.

Reply:
Removed it.